src/utils/utils.cpp

```diff
@@ -630,3 +641,21 @@ std::string mp::utils::emit_cloud_config(const YAML::Node& node)
 {
     return fmt::format("#cloud-config\n{}\n", emit_yaml(node));
 }
+
+mp::id_mappings mp::utils::unique_id_mappings(const mp::id_mappings& xid_mappings)
+{
+    mp::id_mappings ret;
+    std::unordered_set<std::pair<int, int>, id_map_hash> id_set;
+
+    for (const auto& id_map : xid_mappings)
+        if (id_set.insert(id_map).second)
+        {
+            ret.push_back(id_map);
+            mpl::log(mpl::Level::trace, category, fmt::format("Inserting map {}:{}", id_map.first, id_map.second));
+        }
+        else
+            mpl::log(mpl::Level::trace, category,
+                     fmt::format("Not inserting repeated map {}:{}", id_map.first, id_map.second));
```

Comment:
> As you prefer. If you think it is small enough to inline, that's fine too.

I think it's ok to leave the implementation in the header, because moving it to `src` would mean to make a refactor (this is used by the daemon, the SSHFS server and the client). That's why I thought at first it would fit in utils. I implemented it inside the header file.

> How about [std::remove_copy_if](https://en.cppreference.com/w/cpp/algorithm/remove_copy)? You could have a predicate that captured your id_set to do the same insert check.

Right, that will simplify the code.

> As for logging, I would drop the one in line 654 and promote the one in 657 to debug. WDYT?
Right, no need to declare that an ID is correct. I did the change.

Reply:
Sounds good. Let me know when this is ready for a second pass.

Reply:
@ricab thanks, I think it is ready. I used `std::remove_copy_if` as you suggested. It looks better, but the log message avoids us to make the function a one-liner :)

I opted for leaving the function in the header but, instead of inlining it, I used a template. WDYT?